File: ts-sdk/src/sdk/models/operations/postlegalholdpolicies.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostLegalHoldPoliciesRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=filter_ended_at" })
  filterEndedAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=filter_started_at" })
  filterStartedAt?: Date;

  @SpeakeasyMetadata({ data: "json, name=is_ongoing" })
  isOngoing?: boolean;

  @SpeakeasyMetadata({ data: "json, name=policy_name" })
  policyName: string;
}


export class PostLegalHoldPoliciesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostLegalHoldPoliciesRequestBody;
}


export class PostLegalHoldPoliciesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  legalHoldPolicy?: shared.LegalHoldPolicy;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: sdks/python/src/boxsdk/models/operations/post_legal_hold_policies.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import clienterror as shared_clienterror
from ..shared import legalholdpolicy as shared_legalholdpolicy


@dataclass_json
@dataclasses.dataclass
class PostLegalHoldPoliciesRequestBody:
    policy_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('policy_name') }})
    description: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    filter_ended_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('filter_ended_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    filter_started_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('filter_started_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    is_ongoing: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('is_ongoing') }})
    

@dataclasses.dataclass
class PostLegalHoldPoliciesRequest:
    request: Optional[PostLegalHoldPoliciesRequestBody] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    retries: Optional[utils.RetryConfig] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class PostLegalHoldPoliciesResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    client_error: Optional[shared_clienterror.ClientError] = dataclasses.field(default=None)
    legal_hold_policy: Optional[shared_legalholdpolicy.LegalHoldPolicy] = dataclasses.field(default=None)
    
